Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -977,6 +990,8 @@ extension BrowserViewController: WKNavigationDelegate {
                 urlBar.updateURL(tab.displayURL);
                 toolbar?.updateBackStatus(webView.canGoBack)
                 toolbar?.updateFowardStatus(webView.canGoForward)
+                toolbar?.updatePageStatus(isWebPage(webView.URL!))
```

Comment:
Same here (re: optional URLs). Everything else looks good, though, so I can just make these changes now and merge.